cmake/qgis-cmake-wrapper.cmake

```diff
@@ -107,7 +107,10 @@ if(TRUE) # Should possibly have a "static only" check
   if(ANDROID)
     find_library(libdl dl)
     get_filename_component(arch_path ${libdl} DIRECTORY)
-    set(pkgcfg_lib_PC_SPATIALITE_c++ "${arch_path}/libc++.so")
+    set(pkgcfg_lib_PC_SPATIALITE_c++ "${arch_path}/${ANDROID_PLATFORM_LEVEL}/libc++.so")
+    if(NOT EXISTS ${pkgcfg_lib_PC_SPATIALITE_c++})
```

Comment:
@m-kuhn , see failure here (https://github.com/opengisch/QField/actions/runs/3065501184/jobs/4949665264) as to the why this hack was introduced.

Reply:
If the build succeeds it's all good 👍